src/uct/ib/rdmacm/rdmacm_cm_ep.h

```diff
@@ -39,26 +40,49 @@ typedef struct uct_rdmacm_cm_ep {
 } uct_rdmacm_cm_ep_t;
 
 enum {
-    UCT_RDMACM_CM_EP_ON_CLIENT     = UCS_BIT(0),
-    UCT_RDMACM_CM_EP_ON_SERVER     = UCS_BIT(1),
-    UCT_RDMACM_CM_EP_CONNECTED     = UCS_BIT(2),
-    UCT_RDMACM_CM_EP_DISCONNECTING = UCS_BIT(3) /* uct_ep_disconnect was called
-                                                   on the ep. this ep is not
-                                                   necessarily disconnected yet */
+    UCT_RDMACM_CM_EP_ON_CLIENT           = UCS_BIT(0),
+    UCT_RDMACM_CM_EP_ON_SERVER           = UCS_BIT(1),
+    UCT_RDMACM_CM_EP_CONNECTED           = UCS_BIT(2),
+    UCT_RDMACM_CM_EP_DISCONNECTING       = UCS_BIT(3), /* uct_ep_disconnect was
+                                                          called on the ep */
+    UCT_RDMACM_CM_EP_REMOTE_DISCONNECTED = UCS_BIT(4)  /* got disconnect event */
```

Comment:
IMHO this flag should be renamed to indicate the local ep's state. REMOTE_DISCONNECTED doesn't necessarily mean one thing about the local ep. maybe something like - DISCONNECT_CB_INVOKED ?

Reply:
On this level we can operate rdmacm evets

Reply:
what does this mean?

Reply:
We have got disconnect event, cb is user (upper) level thing

Reply:
yes but uct invokes the cb, so it is aware of it